octopus-server/src/main/java/octopus/teamcity/server/generic/CreateDeploymentStep.java

```diff
@@ -0,0 +1,74 @@
+package octopus.teamcity.server.generic;
+
+import java.util.List;
+import java.util.Map;
+
+import com.intellij.openapi.util.text.StringUtil;
+import jetbrains.buildServer.serverSide.InvalidProperty;
+import octopus.teamcity.common.createdeployment.CreateDeploymentPropertyNames;
+import octopus.teamcity.common.createdeployment.CreateDeploymentUserData;
+import org.apache.commons.compress.utils.Lists;
+
+public class CreateDeploymentStep extends OctopusBuildStep {
+
+  private final CreateDeploymentPropertyNames KEYS = new CreateDeploymentPropertyNames();
+
+  public CreateDeploymentStep() {
+    super(
+        "create-deployment",
+        "Create deployment",
+        "editCreateDeploymentParameters.jsp",
+        "viewCreateDeploymentParameters.jsp");
+  }
+
+  @Override
+  public String describeParameters(Map<String, String> properties) {
+    final CreateDeploymentUserData userData = new CreateDeploymentUserData(properties);
+
+    return String.format(
+        "Project: %s\nEnvironments: %s\nRelease version: %s",
```

Comment:
that's pretty nice, way cleaner than the BuildInfo/Common - probably should refactor them (later)